daemon/inertiad/auth/users.go

```diff
@@ -227,3 +229,87 @@ func (m *userManager) IsAdmin(username string) (bool, error) {
 	})
 	return admin, err
 }
+
+// isTOTPEnabled checks if a given user has TOTP enabled
+func (m *userManager) IsTOTPEnabled(username string) (bool, error) {
+	TOTPenabled := false
+
+	err := m.db.View(func(tx *bolt.Tx) error {
+		users := tx.Bucket(m.usersBucket)
+		propsBytes := users.Get([]byte(username))
+		if propsBytes != nil {
+			props := &userProps{}
+			err := json.Unmarshal(propsBytes, props)
+			if err != nil {
+				return errors.New("Corrupt user properties: " + err.Error())
+			}
+			if props.TotpKey != "" {
+				TOTPenabled = true
+			}
+		}
+		return nil
+	})
+	return TOTPenabled, err
+}
+
+// enableTOTP enables TOTP for a user
+func (m *userManager) EnableTOTP(username string) error {
+	err := m.db.Update(func(tx *bolt.Tx) error {
+		users := tx.Bucket(m.usersBucket)
+		propsBytes := users.Get([]byte(username))
+		if propsBytes != nil {
+			props := &userProps{}
+			err := json.Unmarshal(propsBytes, props)
+			if err != nil {
+				return errors.New("Corrupt user properties: " + err.Error())
+			}
+
+			totpKey, totpErr := crypto.GenerateSecretKey(username)
+			if totpErr != nil {
+				return errors.New("Error generating secret totp key: " + totpErr.Error())
+			}
+			props.TOTPBackupCodes = crypto.GenerateBackupCodes()
+			props.TotpKey = totpKey.Secret()
```

Comment:
Im not familiar with the mechanics of TOTP but it might be more apt to call this property of `userProps` `TotpSecret` instead

Reply:
Changed to Totp